### ISSUE TITLE
awk: cmd. line:1: fatal: division by zero attempted

### ISSUE THREAD

**Issue by BeetleMilk** [SAMtoBAM/MUMandCo#25]
I tried running MUM&Co on several query-reference pairs. In several occasions, I got the error:

Matching query and reference chromosomes

awk: cmd. line:1: fatal: division by zero attempted

But, unlike in one of the closed issues here with the same error, the program fails after this. How could I fix this? I noticed that this happened on more evolutionarily distant genome pairs. With more similar genomes, MUM&Co ran fine.

**Comment by SAMtoBAM** [comment]
Hi there,

Do you have the alignment files? Can you have a look at the .coords file and see what is in there?
Or send me them and I will look

If it is due to distance it is probably a problem with the initial alignment

Thanks

**Comment by BeetleMilk** [comment]
The contents of the x_query.delta_filter.coords file:
```
NUCMER

[S1]    [E1]    [S2]    [E2]    [LEN 1] [LEN 2] [% IDY] [LEN R] [LEN Q] [COV R] [COV Q] [FRM]   [TAGS]
20704938        20708279        25738338        25741651        3342    3314    96.68   21046960        28443022        0.02    0.01    1       1       CM002478.1      chr08
NPB_lper_query.delta_filter.coords (END)
```

x_ref.delta_filter.coords:

```
NUCMER

[S1]	[E1]	[S2]	[E2]	[LEN 1]	[LEN 2]	[% IDY]	[LEN R]	[LEN Q]	[COV R]	[COV Q]	[FRM]	[TAGS]
25738338	25741651	20704938	20708279	3314	3342	96.68	28443022	21046960	0.01	0.02	1	1	chr08	CM002478.1
```

**Comment by SAMtoBAM** [comment]
So there is only a 3kb region identified as a match between these two genomes which explains the issue
I would guess therefore that the step that is failing is when it attempts to pair contigs/scaffolds etc it finds there is no pairing which is an unusual case.

Unfortunately this seems to be an issue of distance which is difficult to solve with nucmer alignments

**Comment by BeetleMilk** [comment]
Yes, that makes sense. The x_query.delta file is very small - 288 bytes. It should normally be several Mb for a genome of similar size. So I think the problem starts at the point of generating the delta file. I wonder if this is due to the parameters being too stringent for dissimilar genomes.
Can you maybe recommend a different aligner for evolutionarily distant genomes?

**Comment by SAMtoBAM** [comment]
So nucmer can be adjusted for more distant genomes but you would have to do it yourself and it isn't really clear on how to do it effectively
You can access the nucmer alignment lines in the script if you want to tinker yourself.

Otherwise minimap2 whole genome alignment default options allow for increasing distances such as
`minimap2 -cx asm10`
which allows for up to 10% distance

**Comment by chtthh** [comment]
I would like to ask the blogger whether this error problem has been solved. I hope you can give me a hint. Thank you

**Comment by SAMtoBAM** [comment]
Whether the maximum distance between the two genomes compared has been increased?
Basically no...as I have made no modifications to the default nucmer parameters used by MUM&Co (which only change based on genome size to improve run time).

I don't have much time but I if I do I will allow for the option of providing the alignment files directly. This way you can find the appropriate nucmer alignment parameters for your genomes

**Comment by chtthh** [comment]
Hello blogger, by providing the option to compare files, do you mean to compare the two genomes directly with other software and then run it with mumandco? Thank you for your clear opinion

**Comment by trm76056** [comment]
Hi, I'm responding as I am having a similar problem, though distance between the genomes is not an issue. My .coords files also contain ~4500 lines. The following are the top few lines of each of those for one of my alignments. Do you have any suggestions as to what could be causing this error to occur if I am not dealing with distantly related genomes? I am currently trying to run it separately by chromosome as when I ran the two full genomes it showed a small number of inversions and only for one chromosome, but through other alignment methods we have been able to see that there are more. 

ref.delta_filter.coords
```
NUCMER

[S1]	[E1]	[S2]	[E2]	[LEN 1]	[LEN 2]	[% IDY]	[LEN R]	[LEN Q]	[COV R]	[COV Q]	[FRM]	[TAGS]
3789	3939	22167777	22167620	151	158	79.52	4163716	23731168	0.00	0.00	1	-1	PGA_scaffold4__2_contigs__length_4163716	PGA_scaffold5__7_contigs__length_23731168
3789	3988	22167819	22167628	200	192	79.33	4163716	23731168	0.00	0.00	1	-1	PGA_scaffold4__2_contigs__length_4163716	PGA_scaffold5__7_contigs__length_23731168
3789	3935	22167866	22167703	147	164	80.49	4163716	23731168	0.00	0.00	1	-1	PGA_scaffold4__2_contigs__length_4163716	PGA_scaffold5__7_contigs__length_23731168
```


query.delta_filter.coords

```
NUCMER

[S1]	[E1]	[S2]	[E2]	[LEN 1]	[LEN 2]	[% IDY]	[LEN R]	[LEN Q]	[COV R]	[COV Q]	[FRM]	[TAGS]
296	579	1378954	1379237	284	284	84.27	23731168	4163716	0.00	0.01	1	1	PGA_scaffold5__7_contigs__length_23731168	PGA_scaffold4__2_contigs__length_4163716
305	595	2973966	2974255	291	290	81.91	23731168	4163716	0.00	0.01	1	1	PGA_scaffold5__7_contigs__length_23731168	PGA_scaffold4__2_contigs__length_4163716
398	780	2295165	2295545	383	381	88.63	23731168	4163716	0.00	0.01	1	1	PGA_scaffold5__7_contigs__length_23731168	PGA_scaffold4__2_contigs__length_4163716

```
